src/serializers/addresses-serializer.js

```diff
@@ -0,0 +1,75 @@
+import { Serializer as JsonApiSerializer } from 'jsonapi-serializer';
+import _ from 'lodash';
+
+import { serializerOptions } from 'utils/jsonapi';
+import { openapi } from 'utils/load-openapi';
+import { apiBaseUrl, resourcePathLink, paramsLink } from 'utils/uri-builder';
+
+const addressResourceProp = openapi.components.schemas.AddressResult.properties.data.properties;
+const addressResourceType = addressResourceProp.type.enum[0];
+const addressResourceAttributes = addressResourceProp.attributes.allOf;
+const addressCombinedAttributes = _.merge(
+  addressResourceAttributes[0], addressResourceAttributes[1],
+);
+const addressResourceKeys = _.keys(addressCombinedAttributes.properties);
+
+/**
+ * Some fields need to be massaged before the can be passed to the serializer
+ *
+ * @param {object} rawAddresses raw address data from data source
+ */
+const prepareRawData = (rawAddresses) => {
+  // handle sub objects programmatically
+  _.forEach(rawAddresses, (address) => {
+    _.forEach(address, (value, key) => {
+      const splitKey = key.split('.');
+      if (splitKey.length > 1) {
+        if (!address[splitKey[0]]) {
+          address[splitKey[0]] = {};
+        }
+        address[splitKey[0]][splitKey[1]] = value;
+      }
+    });
```

Comment:
I think in general, when we're taking an object and then modifying both keys and values, we should use `_.transform` or `_.reduce`. I'm not sure if the values are just being moved to different keys or if some of them are kept how they are. If they're all being moved, `_.mapKeys` might work better. Not sure about here though since depth is added.

Actually, if the purpose of this function is just to evaluate `prop1.prop2: val` as `{ prop1: { prop2: val } }`, you could look into other lodash methods like `_.set`. Many of them can evaluate nested paths like this.